src/Microsoft.OData.Core/UriParser/Resolver/AlternateKeysODataUriResolver.cs

```diff
@@ -39,16 +39,21 @@ public AlternateKeysODataUriResolver(IEdmModel model)
         /// <returns>The resolved key list.</returns>
         public override IEnumerable<KeyValuePair<string, object>> ResolveKeys(IEdmEntityType type, IDictionary<string, string> namedValues, Func<IEdmTypeReference, string, object> convertFunc)
         {
-            IEnumerable<KeyValuePair<string, object>> convertedPairs;
-            try
+            IEnumerable<KeyValuePair<string, object>> convertedPairs = null;
+            string[] keyNames = type.DeclaredKey.Select(keyProperty => keyProperty.Name).ToArray();

```

Comment:
Only the declared keys, or keys from base type also?

Reply:
I've replaced with `type.Key()`